backend/database/database.go

```diff
@@ -391,6 +394,54 @@ func (d *Database) ReviewsByVendorID(vendorID uuid.UUID) ([]Review, error) {
 	return result, rows.Err()
 }
 
+//query to find out if the vendor has been starred, it returns true if it is false if not
+func (d *Database) StarredVendorFavorite(ReviewID uuid.UUID, VendorID uuid.UUID) (bool, error) {
+	var starred bool
+	const command = `
+		SELECT *
+		FROM Reviews
+		WHERE ID=?
+		AND VendorID = ?
```

Comment:
This command would actually return the existence of a review, but you want to know if the review exists AND is favorited. Even though the command is wrong, I will approve this, and you can fix it in the future.